components/event/EditForm.vue

```diff
@@ -0,0 +1,397 @@
+    companyUID: null,
+    attendants: null,
+    uid: null,
+  }
+
+  // Set state to event data if data fetched successfully
+  const state = reactive(
+    event.value
+      ? { ...Object.values(embedKeyIntoObjectValues(event.value))[0] }
+      : { ...initialState }
+  )
+
+  // Alert state
+  const initialAlertState = {
+    show: false,
+    alertRoute: '',
+    type: undefined as AlertType,
+  }
+
+  const alertState = reactive({
+    ...initialAlertState,
+  })
+
+  // Show appropriate success alert after signing up for event
+  const displaySuccessAlert = (alertRoute: string) => {
+    alertState.alertRoute = alertRoute
+    alertState.type = 'success'
+    alertState.show = true
+  }
+
+  // Show appropriate error alert for failed API calls
+  const displayErrorAlertFromMessage = (errorMessage: string) => {
+    const content = getAlertContent(errorMessage)
+    alertState.alertRoute = content.alertRoute
+    alertState.type = content.type
+    alertState.show = content.show
+  }
+
+  const form = ref()
+
+  // Boolean control states
+  const editMode = computed(() => !!state.uid || !!state.eventUID)
+  const isLoadingAttendants = ref(false)
+  const hasAttendants = computed(
+    () => Object.hasOwn(state, 'attendants') && !!state.attendants
+  )
+
+  // Get updated event attendants
+  const refreshAttendants = async () => {
+    isLoadingAttendants.value = true
+    const data = await $fetch('/api/event', {
+      query: { eventUID: props.eventUid || state.uid },
+    })
+    if (!data) return
+
+    // Only retain attendants
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { attendants, ...remains } = Object.values(data)[0]
+    state.attendants = attendants
+
+    isLoadingAttendants.value = false
+  }
+
+  // TODO: #184 Add code for removing user from event
+
+  const removeFromEvent = async (userUID: string) => {
+    await $fetch('/api/event/register', {
+      method: 'DELETE',
+      body: {
+        eventUID: props.eventUid || state.uid,
+        userUID,
+      },
+    })
+
+    // Code for removing user
+    refreshAttendants()
+  }
+
+  const handleEmptyStateValues = () => {
+    state.capacity =
+      state.capacity === undefined || Number(state.capacity) === 0
+        ? null
+        : Number(state.capacity)
+
+    state.eventUID = state.uid
+
+    state.location.map = state.location.map ? state.location.map : null
+  }
+
+  const validateForm = async () => {
+    const { valid } = await form.value.validate()
+    if (!valid) throw new Error('Form is not valid')
+  }
+
+  const saveChanges = async () => {
+    validateForm()
+    handleEmptyStateValues()
+    await $fetch('/api/event', {
+      method: 'PUT',
+      body: state,
+    }).catch((error) => displayErrorAlertFromMessage(error.statusMessage))
+
+    displaySuccessAlert('alert.success.event.edit.modified')
+    setTimeout(() => navigateTo('/admin/events'), 2000)
+  }
+
+  const createEvent = async () => {
+    validateForm()
```

Comment:
If I try to create a new event from a completely empty form the api call runs, and the message says a new event has been created even though the api call returned an error, as not all data was defined. Need to catch the error from validateForm()

Reply:
<img width="431" alt="img" src="https://github.com/et-dagen/et-dagen/assets/48122613/bcef7f12-5c6d-40de-b21b-05973b8fa2d1">
Error from validateForm()

Reply:
The .catch on the api call runs but the `displayErrorAlertFromMessage` does not display any message